### PR TITLE
Adding documentation for necessary GitHub and GitLab permissions for …

### PR DESCRIPTION
**Pull request by jsmoxon**
…Assistant

# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Review by s-santillan — APPROVED** [review]

**Comment by s-santillan** [comment]
Hi @armchairlinguist I am tagging you mostly for awareness so you know that Jack's content is making its way into the docs and any notes you may have of course.
 
The content in the FAQ portion of the draft will be added in a separate PR and formatted for docs (i.e. not question and answer, but same content).

Thank you!

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 1]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
 * Semgrep Inc stores and retains these code snippets for up to 6 months. Semgrep Inc. will update you with at least a 30-day notice, if we make any changes to the retention policy.
 * Semgrep Inc is a paying customer of OpenAI and has a Data Protection Agreement signed with them (provided upon request by contacting [support@semgrep.com](mailto:support@semgrep.com). The code snippets we upload will be persisted by OpenAI temporarily, following their data usage policies at [Open AI API data Usage Policies](https://openai.com/policies/api-data-usage-policies).
 * For more details, see the [Semgrep Assistant FAQ](https://get.semgrep.dev/assistant).
 
+## Required GitHub or GitLab repository permissions
+
+Semgrep Assistant requires the same permissions that Semgrep needs to integrate with GitHub or GitLab such as permissions that allow Semgrep to run CI jobs and post comments to PRs or MRs. See [<i class="fa-regular fa-file-lines"></i> Requested permissions for GitHub and GitLab](/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab) for a list of permissions.
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:67]:
```suggestion
Semgrep Assistant requires the same permissions that Semgrep needs to integrate with GitHub or GitLab, such as permissions that allow Semgrep to run CI jobs and post comments to PRs or MRs. See [<i class="fa-regular fa-file-lines"></i> Requested permissions for GitHub and GitLab](/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab) for a list of permissions.
```

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 2]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
+    defaultValue="github"
+    values={[
+    {label: 'GitHub', value: 'github'},
+    {label: 'GitLab', value: 'gitlab'},
+    ]}
+>
+
+<TabItem value='github'>
+
+Semgrep Assistant requires [read access to your code in GitHub](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28). This is done through installation of a private Semgrep GitHub app that you install during Assistant setup. This private Semgrep GitHub app:
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:81]:
```suggestion
Semgrep Assistant requires [read access to your code in GitHub](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28). This is done through a private Semgrep GitHub app that you install during Assistant setup. This private Semgrep GitHub app:
```

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 3]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html). 
+
+* You can revoke [project level access tokens](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#revoke-a-project-access-token) or [personal access tokens](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#revoke-a-personal-access-token) at any time. 
+* Semgrep Assistant only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* The token can be configured to limit its scope to specific projects or individuals. You do not need to give read access to all repositories in your GitLab organization).
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:95]: Should we use the term "project" throughout, here, rather than switching to "repositories"?

Comment (s-santillan) [reply docs/semgrep-code/semgrep-assistant-code.md:95]: I can do that. :)

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 4]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
+
+* Is fully under your control so you can revoke access or specific permissions at any time by visiting **Settings > Applications** in GitHub.
+* Only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* Can be configured to limit its scope to specific repositories. You do not need to give read access to all repositories in your GitHub organization. 
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html).
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:91]: I noticed when I was going over this again that we say "self-managed GitLab" here but I think I've usually seen us say "GitLab Self-Managed". However, we do use 'self-hosted' as a prefixing generic in a few places so...whatever makes the most sense!

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 5]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
+* Only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* Can be configured to limit its scope to specific repositories. You do not need to give read access to all repositories in your GitHub organization. 
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html). 
+
+* You can revoke [project level access tokens](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#revoke-a-project-access-token) or [personal access tokens](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#revoke-a-personal-access-token) at any time.
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:93]:
```suggestion
* You can revoke [project access tokens](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#revoke-a-project-access-token) or [personal access tokens](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#revoke-a-personal-access-token) at any time. 
```
GL doesn't seem to use "level" there, and we don't use it above; it feels like it should be hyphenated if we do use it, so I vote for removing it.

File: docs/semgrep-code/semgrep-assistant-code.md [hunk 6]
```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
...
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html). 
+
+* You can revoke [project level access tokens](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#revoke-a-project-access-token) or [personal access tokens](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#revoke-a-personal-access-token) at any time. 
+* Semgrep Assistant only accesses source code repositories (projects) on a file-by-file basis; it does not need or request org-level access to your codebase.
+* The token can be configured to limit its scope to specific projects or individuals. You do not need to give read access to all projects in your GitLab organization).
```
Comment (armchairlinguist) [inline docs/semgrep-code/semgrep-assistant-code.md:95]:
```suggestion
* The token can be configured to limit its scope to specific projects or individuals. You do not need to give read access to all projects in your GitLab organization.
```

**Review by armchairlinguist — APPROVED** [review]
Looks good. I had some small nits from reading over it more carefully (sorry, should have caught that stuff before) but definitely optional.

File: docs/semgrep-code/semgrep-assistant-code.md (hunk 4 above)
Comment (s-santillan) [reply docs/semgrep-code/semgrep-assistant-code.md:91]: "GitLab or self-managed GitLab" does sound a bit weird. To make it flow a little better: "Semgrep Assistant requires the API scope to run in both cloud and self-managed GitLab instances."

**Pull request merged**